test/e2e/major_upgrade_v2.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/celestiaorg/celestia-app/v2/test/txsim"
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/tendermint/tendermint/rpc/client/http"
+)
+
+func MajorUpgradeToV2(logger *log.Logger) error {
+	latestVersion, err := testnet.GetLatestVersion()
+	testnet.NoError("failed to get latest version", err)
+
+	logger.Println("Running major upgrade to v2 test", "version", latestVersion)
+
+	numNodes := 4
+	upgradeHeight := int64(12)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger.Println("Creating testnet")
+	testNet, err := testnet.New("runMajorUpgradeToV2", seed, nil, "test")
+	testnet.NoError("failed to create testnet", err)
+
+	defer testNet.Cleanup()
+
+	testNet.SetConsensusParams(app.DefaultInitialConsensusParams())
+
+	preloader, err := knuu.NewPreloader()
+	testnet.NoError("failed to create preloader", err)
+
+	defer func() { _ = preloader.EmptyImages() }()
+	testnet.NoError("failed to add image", preloader.AddImage(testnet.DockerImageName(latestVersion)))
+
+	logger.Println("Creating genesis nodes")
+	for i := 0; i < numNodes; i++ {
+		err := testNet.CreateGenesisNode(latestVersion, 10000000, upgradeHeight, testnet.DefaultResources)
+		testnet.NoError("failed to create genesis node", err)
+	}
+
+	kr, err := testNet.CreateAccount("alice", 1e12, "")
+	testnet.NoError("failed to create account", err)
+	// start the testnet
+
+	logger.Println("Setting up testnet")
+	testnet.NoError("Failed to setup testnet", testNet.Setup())
+	logger.Println("Starting testnet")
+	testnet.NoError("Failed to start testnet", testNet.Start())
+
+	errCh := make(chan error)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	opts := txsim.DefaultOptions().WithSeed(seed).SuppressLogs()
+	sequences := txsim.NewBlobSequence(txsim.NewRange(200, 4000), txsim.NewRange(1, 3)).Clone(5)
+	sequences = append(sequences, txsim.NewSendSequence(4, 1000, 100).Clone(5)...)
+	go func() {
+		errCh <- txsim.Run(ctx, testNet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
+	}()
+
+	// assert that the network is initially running on v1
+	heightBefore := upgradeHeight - 1
+	for i := 0; i < numNodes; i++ {
+		client, err := testNet.Node(i).Client()
+		testnet.NoError("failed to get client", err)
+
+		testnet.NoError("failed to wait for height", waitForHeight(testNet, testNet.Node(i), upgradeHeight, time.Minute))
+
+		resp, err := client.Header(ctx, &heightBefore)
+		testnet.NoError("failed to get header", err)
+		logger.Println("Node", i, "is running on version", resp.Header.Version.App)
+		if resp.Header.Version.App != v1.Version {
+			return fmt.Errorf("version mismatch before upgrade: expected %d, got %d", v1.Version, resp.Header.Version.App)
+		}
+
+		resp, err = client.Header(ctx, &upgradeHeight)
+		testnet.NoError("failed to get header", err)
+		if resp.Header.Version.App != v2.Version {
+			return fmt.Errorf("version mismatch before upgrade: expected %d, got %d", v2.Version, resp.Header.Version.App)
```

Comment:
I think this error message is wrong

```suggestion
			return fmt.Errorf("version mismatch after upgrade: expected %d, got %d", v2.Version, resp.Header.Version.App)
```